### PR TITLE
Docs: add an example `Object.assign()` for rule no-import-assign

### PR DESCRIPTION
**Pull request by aladdin-add**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
as https://github.com/eslint/eslint/issues/14911#issuecomment-896276484, it has been supported but not documented.

### CODE REVIEW

File: docs/rules/no-import-assign.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ mod = 1          // ERROR: 'mod' is readonly.
 named = 2        // ERROR: 'named' is readonly.
 mod_ns.named = 3 // ERROR: the members of 'mod_ns' is readonly.
 mod_ns = {}      // ERROR: 'mod_ns' is readonly.
+Object.assign(mod_ns, { foo: "foo" }) // ERROR: 'mod_ns' is not extensible.
```
Comment (snitin315) [inline docs/rules/no-import-assign.md:21]:
```suggestion
Object.assign(mod_ns, { foo: "foo" }) // ERROR: The members of 'mod_ns' are readonly.
```

Comment (snitin315) [reply docs/rules/no-import-assign.md:21]: As shown in [Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWltcG9ydC1hc3NpZ246IFwiZXJyb3JcIiAqL1xuXG5pbXBvcnQgKiBhcyBtb2RfbnMgZnJvbSBcIi4vbW9kLm1qc1wiXG5cbk9iamVjdC5hc3NpZ24obW9kX25zLCB7IGZvbzogXCJmb29cIiB9KTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

File: docs/rules/no-import-assign.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ mod = 1          // ERROR: 'mod' is readonly.
 named = 2        // ERROR: 'named' is readonly.
 mod_ns.named = 3 // ERROR: the members of 'mod_ns' is readonly.
```
Comment (snitin315) [inline docs/rules/no-import-assign.md:19]:
```suggestion
mod_ns.named = 3 // ERROR: The members of 'mod_ns' are readonly.
```

Comment (snitin315) [reply docs/rules/no-import-assign.md:19]: Let's take the opportunity to fix this as well.

Comment (aladdin-add) [reply docs/rules/no-import-assign.md:19]: 👍

File: docs/rules/no-import-assign.md (hunk 1 above)
Comment (aladdin-add) [reply docs/rules/no-import-assign.md:21]: but in this case, it was to extend `mod_ns`, not to write the members.

Comment (snitin315) [reply docs/rules/no-import-assign.md:21]: Extending is the same as writing new members. I think we should not change the Error message here as it is coming from ESLint, tough we can add a comment above to show it's not extensible:

```suggestion
// Can't extend 'mod_ns'
Object.assign(mod_ns, { foo: "foo" }) // ERROR: The members of 'mod_ns' are readonly.
```

Comment (aladdin-add) [reply docs/rules/no-import-assign.md:21]: makes sense! 👍

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**